homeassistant/components/netgear/__init__.py

```diff
@@ -52,14 +55,37 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         configuration_url=f"http://{entry.data[CONF_HOST]}/",
     )
 
+    async def async_update_data():
+        """Fetch data from the router."""
+        await router.async_update_device_trackers()
+        return
+
+    # Create update coordinator
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=router.device_name,
+        update_method=async_update_data,
+        update_interval=SCAN_INTERVAL,
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data[DOMAIN][entry.unique_id] = {
+        KEY_ROUTER: router,
+        KEY_COORDINATOR: coordinator,
+    }
+
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
 
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    unload_ok = await hass.config_entries.async_unload_platforms(
+        entry, PLATFORMS_UNLOAD
```

Comment:
The router module is not a valid platform to unload.

Reply:
fixed